File: examples/showcase.ml
struct Item {
  name: string;
  qty: int;
  price: float;
}

func total(items: Item[]): float {
  let sum: float = 0.0;
  for (let i: int = 0; i < len(items); i = i + 1) {
    sum = sum + items[i].price * float(items[i].qty);
  }
  return sum;
}

func tag(it: Item): string {
  return it.name + " x" + it.qty + (it.qty > 1 ? " (bulk)" : "");
}

func main(): void {
  let items: Item[] = [Item("anvil", 2, 19.5), Item("tongs", 1, 7.25), Item("flux", 6, 0.5)];
  for (let i: int = 0; i < len(items); i = i + 1) {
    if (items[i].qty == 0) { continue; }
    print(tag(items[i]));
  }
  print(total(items));
  print(int(total(items)));
  print(substr("=== FOUNDRY ===", 4, 11));
  print(len(items) >= 3);
}

File: showcase/showcase.php
<?php

function __f(float $x): string {
    $s = rtrim(number_format($x, 6, ".", ""), "0");
    if (substr($s, -1) === ".") $s .= "0";
    return $s;
}

class Item {
    public function __construct(public $name, public $qty, public $price) {}
}

function total(&$items) {
    $sum = 0.0;
    for ($i = 0; ($i < count($items)); $i = ($i + 1)) {
        $sum = ($sum + ($items[$i]->price * (float)($items[$i]->qty)));
    }
    return $sum;
}

function tag($it) {
    return ((($it->name . ' x') . $it->qty) . ((($it->qty > 1)) ? ' (bulk)' : ''));
}

function main() {
    $items = [new Item('anvil', 2, 19.5), new Item('tongs', 1, 7.25), new Item('flux', 6, 0.5)];
    for ($i = 0; ($i < count($items)); $i = ($i + 1)) {
        if (($items[$i]->qty === 0)) {
            continue;
        }
        echo tag($items[$i]), "\n";
    }
    echo __f(total($items)), "\n";
    echo (int)(total($items)), "\n";
    echo substr('=== FOUNDRY ===', 4, (11) - (4)), "\n";
    echo ((count($items) >= 3)) ? "true" : "false", "\n";
}

main();
